Objects/listobject.c

```diff
@@ -2717,6 +2717,35 @@ list___init___impl(PyListObject *self, PyObject *iterable)
     return 0;
 }
 
+static PyObject *
+list_vectorcall(PyObject *type, PyObject * const*args,
+                size_t nargsf, PyObject *kwnames)
+{
+    if (kwnames && PyTuple_GET_SIZE(kwnames) != 0) {
+        PyErr_Format(PyExc_TypeError, "list() takes no keyword arguments");
+        return NULL;
+    }
+    Py_ssize_t nargs = PyVectorcall_NARGS(nargsf);
+    if (nargs > 1) {
+        PyErr_Format(PyExc_TypeError, "list() expected at most 1 argument, got %zd", nargs);
+        return NULL;
+    }
+
+    assert(PyType_Check(type));
```

Comment:
Can you move this assert into the top of the function body since this parameter is the 1st parameter?

Reply:
I don't see why that's necessary.